### PR TITLE
Add Rust-Analyzer reload workspace command

### PR DESCRIPTION
**Pull request by teozkr**
This lets Rust-Analyzer pick up project changes without a full server restart.

### CODE REVIEW

File: clients/lsp-rust.el [hunk 1]
```diff
@@ -604,6 +604,11 @@ them with `crate` or the crate name they refer to."
          (result (lsp-send-request (lsp-make-request "experimental/joinLines" params))))
     (lsp--apply-text-edits result 'code-action)))
 
+(defun lsp-rust-analyzer-reload-workspace ()
+  "Reload workspace, picking up changes from Cargo.toml"
+  (interactive)
+  (lsp-send-request (lsp-make-request "rust-analyzer/reloadWorkspace")))
```
Comment (ericdallo) [inline clients/lsp-rust.el:610]:
```suggestion
  (lsp--cur-workspace-check)
  (lsp-send-request (lsp-make-request "rust-analyzer/reloadWorkspace")))
```

**Review by ericdallo — APPROVED** [review]
LGTM

**Comment by teozkr** [comment]
https://github.com/emacs-lsp/lsp-mode/runs/2148282265 looks more like a test infra problem than something related to this PR, as far as I can see?

**Comment by ericdallo** [comment]
Yes an out of memory error it seems, I retriggered the CI

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**